### PR TITLE
issue_19

### PR DESCRIPTION
**Pull request by worldomonation**
Made multiple changes, not all of them complete or as modularized as I like.

- setup: removed script entry point for adr-query.
- adr.cli: removed cli() as the entry point for recipe.
- adr.cli: copied some code from adr.query; ultimate goal is to encapsulate all query calls.
- adr.cli: created new method main() which is the new entry point for both `adr query` and `adr recipe`/
- adr.cli: switched to using subparsers to support the above.
- adr.cli: extracted common calls between `recipe` and `query` to helper functions.
- adr.query: changed log import from adr.cli to reference by name.

### CODE REVIEW

**Comment by worldomonation** [comment]
Carried on with additional refactoring on master branch in the forked repository:

- setup: removed `adr-query` entry point
- adr.cli: removed query handler that was temporarily copied to `adr.cli:query_handler` for changeset `c53fe73`
- adr.query: refactored how the query file is found and read. Now it assumes the name exists, due to previous checks being present.
- adr.query: `run_query` (old) renamed to `load_and_run_query`; code logic remains unchanged.
- adr.query: new function `run_query` created to mirror `run_recipe`. It wraps around the top level business logic of the old `cli` method.
- adr.query: marked older methods as deprecated.
- adr.query: added docstrings.

**Comment by ahal** [comment]
Please rebase this on the latest master, I added a whole bunch of new ARGUMENT_GROUPS and it looks like these might be lost.

**Comment by worldomonation** [comment]
Incorporated upstream changes, moved the new `ARGUMENT_GROUPS` from cli.py to recipe.py like in the refactor.

File: adr/query.py [hunk 1]
```diff
@@ -13,15 +13,34 @@
...
 
+log = logging.getLogger('adr')
 here = os.path.abspath(os.path.dirname(__file__))
 
 ACTIVE_DATA_URL = 'http://activedata.allizom.org/query'
 QUERY_DIR = os.path.join(here, 'queries')
+FAKE_CONTEXT = {
+    'branch': 'mozilla-central',
+    'branches': ['mozilla-central'],
+    'from_date': 'now-week',
```
Comment (klahnakoski) [inline adr/query.py]: do not use `now`, use `today`

**Pull request closed without merging**